WooCommerce/Classes/ViewRelated/MainTabBarController.swift

```diff
@@ -842,11 +842,15 @@ private extension MainTabBarController {
         }
         hubMenuTabCoordinator?.activate(siteID: siteID)
 
-        // Set dashboard to be the default tab - disable optional tabs by default.
-        selectedIndex = WooTab.myStore.visibleIndex(isPOSTabVisible: false,
-                                                    isBookingsTabVisible: false)
-        updateTabViewControllers(isPOSTabVisible: false,
-                                 isBookingsTabVisible: false)
+        // Sets dashboard to be the default tab.
+        selectedIndex = WooTab.myStore.visibleIndex(isPOSTabVisible: isPOSTabVisible,
+                                                    isBookingsTabVisible: isBookingsTabVisible)
+
+        // Updates site ID for the POS coordinator to ensure correct data
+        posTabCoordinator?.didSwitchStore(id: siteID)
```

Comment:
I'm hoping the siteID can remain constant in `POSTabCoordinator`. How about reinitializing the `POSTabCoordinator` here instead of updating the siteID? Would it cause any issues?

Reply:
I'd consider separating the tab visibility logic in `POSEntryPointEligibilityCheckerProtocol` to a new protocol that stays in the app, so that `POSEntryPointEligibilityCheckerProtocol` just has `checkEligibility` and `refreshEligibility` that should not depend on the full `Site` (the CIAB check is performed on the tab visibility).

We're also in the middle of moving POS code into a separate module, and I'm refactoring the dependencies of `PointOfSaleEntryPointView` passed in `POSTabCoordinator`. Please go ahead with merging this PR and I'll look into the refactoring as part of the POS module refactoring work.